crates/infra/cli/src/commands/completions/mod.rs

```diff
@@ -0,0 +1,23 @@
+use anyhow::Result;
+use clap::{CommandFactory, Parser};
+use clap_complete::Generator;
+
+/// Generate shell completions for this CLI.
+#[derive(Clone, Debug, Parser)]
+pub struct CompletionController {
+    /// The shell to generate the completions for
+    #[arg(value_enum)]
+    shell: clap_complete::Shell,
```

Comment:
I suggest making this a vararg, same as other commands like `check`, `test`, and `lint`, and using `LintCommand::execute_in_order()` to generate all shells in one go if the user wants.

Reply:
Oh. Brain moment! I misunderstood the intention behind this controller. Sounds good.